nikola/plugins/command/serve.py

```diff
@@ -144,7 +145,16 @@ def _execute(self, options, args):
                 handler_factory = OurHTTPRequestHandler
             else:
                 handler_factory = _create_RequestHandler_removing_basepath(base_path)
-            self.httpd = OurHTTP((options['address'], options['port']), handler_factory)
+
+            try:
+                self.httpd = OurHTTP((options['address'], options['port']), handler_factory)
+            except OSError as e:
+                if e.errno == errno.EADDRINUSE:
+                    print(f"Port address {options['port']} already in use, please use "
+                          f"the `-p <port>` option to select a different one.")
+                    sys.exit(1)
```

Comment:
I would like to see the `test_dev_server_server` augmented to verify this works as intended.

But doing `exit` makes this improvement much harder to test than it would otherwise be.

Reply:
After looking into this a bit more (and opening https://github.com/getnikola/nikola/issues/3828 for another issue we have with this), I found all that's needed here is a humble `return 3` instead of the `exit`.